app/scripts/metamask-controller.js

```diff
@@ -1288,6 +1288,9 @@ export default class MetamaskController extends EventEmitter {
       updateAndApproveTransaction: txController.updateAndApproveTransaction.bind(
         txController,
       ),
+      trackTransactionMetricsEvent: txController.trackTransactionMetricsEvent.bind(
```

Comment:
[REQ] We should not expose this function to the UI. I added the public method createTransactionEventFragment for the UI to verify that the proceeding fragment exists

Reply:
It will be useful for UI to be able to re-use this whole function otherwise I need to duplicate some of this code in UI. Is it ok if I add a method on top of this one for use by UI ?